Makefile

```diff
@@ -1,131 +1,44 @@
 # setup (use XXX=<value> make <target> to override)
-GCR_URL ?= us.gcr.io/vcm-ml
-COMMIT_SHA := $(shell git rev-parse HEAD)
-DOCKERFILE ?= docker/Dockerfile
-ENVIRONMENT_TAG_NAME ?= latest
-COMPILE_OPTION ?=
-COMPILE_TARGET ?= fv3gfs-compiled
-BUILD_ARGS ?=
-BUILD_FROM_INTERMEDIATE ?= n
-ENVIRONMENT_TARGET ?= fv3gfs-environment
-CUDA ?= n
-OTHER_MOUNTS ?= 
-
-# base images w/ or w/o CUDA
-ifeq ($(CUDA),n)
-	BASE_IMAGE ?=ubuntu:18.04
-	DEP_TAG_NAME ?=gnu7-mpich314-nocuda
-else
-	BASE_IMAGE ?=nvidia/cuda:10.2-devel-ubuntu18.04
-	DEP_TAG_NAME ?=gnu7-mpich314-cuda102
-endif
-BUILD_ARGS += --build-arg BASE_IMAGE=$(BASE_IMAGE)
-
-# image names (use XXX_IMAGE=<name> make <target> to override)
-COMPILED_TAG_NAME ?=$(DEP_TAG_NAME)
-COMPILED_IMAGE ?= $(GCR_URL)/$(COMPILE_TARGET):$(COMPILED_TAG_NAME)
-SERIALIZE_IMAGE ?= $(GCR_URL)/$(COMPILE_TARGET):$(COMPILED_TAG_NAME)-serialize
-EMULATION_IMAGE ?= $(GCR_URL)/$(COMPILE_TARGET):$(COMPILED_TAG_NAME)-emulation
-ENVIRONMENT_IMAGE ?= $(GCR_URL)/$(ENVIRONMENT_TARGET):$(ENVIRONMENT_TAG_NAME)
-MPI_IMAGE ?= $(GCR_URL)/mpi-build:$(DEP_TAG_NAME)
-FMS_IMAGE ?= $(GCR_URL)/fms-build:$(DEP_TAG_NAME)
-ESMF_IMAGE ?= $(GCR_URL)/esmf-build:$(DEP_TAG_NAME)
-SERIALBOX_IMAGE ?= $(GCR_URL)/serialbox-build:$(DEP_TAG_NAME)
-
-# used to shorten build times in CircleCI
-ifeq ($(BUILD_FROM_INTERMEDIATE),y)
-	BUILD_ARGS += --build-arg FMS_IMAGE=$(FMS_IMAGE) --build-arg ESMF_IMAGE=$(ESMF_IMAGE) --build-arg SERIALBOX_IMAGE=$(SERIALBOX_IMAGE) --build-arg MPI_IMAGE=$(MPI_IMAGE)
-endif
-
-.PHONY: help build build_environment build_compiled build_serialize build_debug
-.PHONY: build_deps push_deps pull_deps enter enter_serialize test update_circleci_reference clean
+.PHONY: help build build_repro build_debug build_wrapper clean
 
 help:
 	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z0-9_-]+:.*?## / {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
 
-build: build_compiled ## build default container image (production)
-
-build_environment: ## build environment container image
-	docker build -f $(DOCKERFILE) -t $(ENVIRONMENT_IMAGE) \
-	--target $(ENVIRONMENT_TARGET) .
-
-build_compiled: ## build production container image
-	docker build \
-		--build-arg compile_option=$(COMPILE_OPTION) \
-		$(BUILD_ARGS) \
-		-f $(DOCKERFILE) \
-		-t $(COMPILED_IMAGE) \
-		--target $(COMPILE_TARGET) .
-
-build_debug: ## build container image for debugging
-	COMPILED_TAG_NAME=$(COMPILED_TAG_NAME)-debug COMPILE_OPTION="REPRO=\\\nDEBUG=Y" $(MAKE) build
-
-build_serialize: ## build container image for serialization
```

Comment:
I guess we no longer have a use case for generating serialized data this way? Maybe worth mentioning that we've deprecated the ability in the documentation?

Reply:
I believe so.  In working on #364, after discussion with @elynnwu and @oliverwm1, we decided to remove `serialbox` from the build / test system since it was no longer needed and would require changes to run in a new base image (as an aside, the deadline we thought we had that triggered the need for that PR seems to have passed without any ill effects).  

In principle, we could keep the docker image around and only use it as a regression test of the serialization capability (say on merges to main, not on each commit to each PR).  That would cordon off the docker image in a place that we would generally not need to interact with it during everyday development, which is the main goal of this PR.  Admittedly, it would probably be good to get @oelbert's opinion on this (happy to chat offline to provide more context), since he is the main person that still might be interested in this functionality (though he may be working in forks, in SHiELD, or off specific earlier commits of this repository, which may make this less relevant).

I agree, whatever we decide, it should be made clear in the top-level README.

Reply:
I spoke with @oelbert about this offline.  He still uses the serialize functionality in the docker image on occasion, but is OK with us removing it, since it is fine for him to refer to older commits of the repository (it is not important for him to keep up with fortran changes that we make).

I will add a note of this deprecation to the README, and point to the last valid commit.